autogluon/utils/tabular/ml/trainer/model_presets/presets.py

```diff
@@ -1,33 +1,148 @@
+import copy
 import logging
-import mxnet as mx
+from collections import defaultdict
+
 from sklearn.linear_model import LogisticRegression, LinearRegression
 
-from ...constants import BINARY, MULTICLASS, REGRESSION, SOFTCLASS
+from ...constants import AG_ARGS, AG_ARGS_FIT, BINARY, MULTICLASS, REGRESSION, SOFTCLASS, PROBLEM_TYPES_CLASSIFICATION
+from ...models.abstract.abstract_model import AbstractModel
 from ...models.lgb.lgb_model import LGBModel
-from ...models.lgb.hyperparameters.parameters import get_param_baseline_custom
 from ...models.lr.lr_model import LinearModel
 from ...models.tabular_nn.tabular_nn_model import TabularNeuralNetModel
 from ...models.rf.rf_model import RFModel
 from ...models.knn.knn_model import KNNModel
 from ...models.catboost.catboost_model import CatboostModel
-from .presets_rf import rf_classifiers, xt_classifiers, rf_regressors, xt_regressors
-from ....metrics import soft_log_loss, mean_squared_error
+from ...models.xt.xt_model import XTModel
+from ....metrics import soft_log_loss
 
 logger = logging.getLogger(__name__)
 
+# Higher values indicate higher priority, priority dictates the order models are trained for a given level.
+DEFAULT_MODEL_PRIORITY = dict(
+    RF=100,
+    XT=90,
+    KNN=80,
+    GBM=70,
+    CAT=60,
+    NN=50,
+    LR=40,
+    custom=0,
```

Comment:
Can wait for future PR/testing, but I think this can be improved with order like:
RF, GBM, CAT, NN, KNN, XT, LR

Reply:
It is likely a different order will perform better, just need to test.